neco-server-exporter/pkg/collector/bpf/collector.go

```diff
@@ -0,0 +1,135 @@
+package bpf
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io/fs"
+	"strconv"
+
+	"github.com/cilium/cilium/pkg/client"
+	"github.com/cilium/ebpf"
+	"github.com/cybozu/neco-containers/neco-server-exporter/pkg/exporter"
+)
+
+type collector struct {
+	ciliumClient *client.Client
+}
+
+var _ exporter.Collector = &collector{}
+
+func NewCollector() exporter.Collector {
+	return &collector{}
+}
+
+func (c *collector) Name() string {
+	return "bpf"
+}
+
+func (c *collector) Collect(ctx context.Context) ([]*exporter.Metric, error) {
+	if err := CheckBPFStatsEnabled(); err != nil {
+		return nil, err
+	}
+	if c.ciliumClient == nil {
+		var err error
+		if c.ciliumClient, err = client.NewClient(""); err != nil {
+			return nil, fmt.Errorf("failed to open Cilium socket: %w", err)
+		}
+	}
+
+	tcxMeta, err := CollectTCXMetadata()
+	if err != nil {
+		return nil, fmt.Errorf("failed to read TCX info: %w", err)
+	}
+
+	endpointMeta, err := CollectEndpointMetadata(c.ciliumClient)
+	if err != nil {
+		// maybe disconnected? connect again on the next iteration
+		c.ciliumClient = nil
+		return nil, fmt.Errorf("failed to read from Cilium socket: %w", err)
+	}
+
+	ret := make([]*exporter.Metric, 0)
+	var id ebpf.ProgramID
+
+ProgramLoop:
```

Comment:
This `ProgramLoop` label is unnecessary I think since there's only a single for loop.

Reply:
Ah, you're right! My comment was incorrect.